compiler-core/src/build/project_compiler.rs

```diff
@@ -397,7 +397,7 @@ where
     ) -> Result<(), Error> {
         for path in self.io.gleam_metadata_files(&build_dir) {
             let reader = BufReader::new(self.io.reader(&path)?);
-            let module = metadata::Metadata::decode(self.ids.clone(), reader)?;
+            let module = metadata::Metadata::decode(self.ids.clone(), reader.buffer())?;
```

Comment:
I think this is taking just the buffer and not the full contents. I would remove the use of a reader and read the entire file.

Reply:
updated